features/spidertron_group_control.lua

```diff
@@ -0,0 +1,116 @@
+        return false
+    end
+
+    if cursor_stack.tile_selection_mode ~= defines.deconstruction_item.tile_selection_mode.never then
+        return false
+    end
+
+    local filters = cursor_stack.entity_filters
+    if #filters ~= 1 or filters[1] ~= 'sand-rock-big' then
+        return false
+    end
+
+      -- check if the player has given the decon planner an icon. This is how  we will determine  their intention
+    if not cursor_stack.blueprint_icons or not cursor_stack.blueprint_icons[1] or not cursor_stack.blueprint_icons[1].signal.name then
+        return false
+    end
+
+    local icon_name = cursor_stack.blueprint_icons[1].signal.name
+    if icon_name ~= "spidertron" then
+        return false
+    end
+
+    return true
+end
+
+Event.add(defines.events.on_player_deconstructed_area , function(event)
+    local player = game.get_player(event.player_index)
+    local cursor_stack = player.cursor_stack
+    if not player or not player.valid then
+        return
+    end
+
+    -- check they actually have a decon planner in their cursor that is setup to be a targetting deconstruction planner.
+    if not is_targetting_deconstruction_planner(cursor_stack) then
+        return
+    end
+
+    local left_top = event.area.left_top
+    local right_bottom = event.area.right_bottom
+    local width = math.abs(left_top.x -  right_bottom.x)
+    local height = math.abs(left_top.y -  right_bottom.y)
+
+    -- A single (small area) click is interpretted as instruction to assign a group of spiders to follow a spider
+    if width <= 1 and height <= 1  then
+        local spidertrons = spider_army[player.name]
+        local target_spider = player.surface.find_entities_filtered{name="spidertron", area={{left_top.x-1,left_top.y-1}, {right_bottom.x+1,right_bottom.y+1}}, limit=1}
+        if #target_spider == 1 then
+            for i, spidertron in pairs(spidertrons) do
+                spidertron.follow_target = target_spider[1]
+            end
+            spider_army[player.name] = {} -- clear spidertrons from table once they've been assigned to follow another spidey lad
+            cursor_stack.label = "Select a group of spidertrons that belong to you! 0 selected."
+        else
+            player.surface.create_entity {
+                name = 'flying-text',
+                text = {'spidertron_group_control.none_found'},
+                position = left_top,
+                render_player_index = player.index
+            }
+        end
+    else -- else the area is bigger than 1x1 and so we assume the player is selecting which spiders to assign
+        local spidertrons = player.surface.find_entities_filtered{area = {left_top, right_bottom}, name="spidertron"}
+        local spidertrons_valid = {}
+        if #spidertrons > 0 then
+            for i, spidertron in pairs(spidertrons) do
+                if spidertron.last_user.name == player.name then
+                    spidertrons_valid[i] = spidertron
```

Comment:
```suggestion
                    spidertrons_valid[#spidertrons_valid + 1] = spidertron
```
We can't use `i` because it could create gaps in the `spidertrons_valid` array which would cause the count to be incorrect.

Reply:
So if I do:
local test = {}
test[1] = 7
test[709] = 7

Then #test is 709 ?

But if I do 
local test = {}
test["one"] = 7
test["709"] = 7

Then #test is 2?

*sigh* I hate lua

Reply:
> So if I do:
local test = {}
test[1] = 7
test[709] = 7
Then #test is 709 ?

`#test` would be 1

> But if I do
local test = {}
test["one"] = 7
test["709"] = 7
Then #test is 2?

#test would be 0.

`#` is the number of consecutive entries from 1 .. n.
0, negative numbers, string and other keys are ignored.

Reply:
Good to know. Thank you.